DeliBuddy/app/src/main/java/yapp/android1/delibuddy/ui/address/detail/AddressDetailViewModel.kt

```diff
@@ -0,0 +1,95 @@
+import yapp.android1.delibuddy.base.BaseViewModel
+import yapp.android1.delibuddy.base.RetryAction
+import yapp.android1.delibuddy.model.Event
+import yapp.android1.delibuddy.util.EventFlow
+import yapp.android1.delibuddy.util.MutableEventFlow
+import yapp.android1.domain.NetworkResult
+import yapp.android1.domain.entity.Address
+import yapp.android1.domain.entity.NetworkError
+import yapp.android1.domain.interactor.usecase.CoordToAddressUseCase
+import javax.inject.Inject
+
+@HiltViewModel
+class AddressDetailViewModel @Inject constructor(
+    private val coordToAddressUseCase: CoordToAddressUseCase
+) : BaseViewModel<Event>() {
+    private var job: Job? = null
+
+    private val _addressResult = MutableStateFlow<Address?>(null)
+    val addressResult: StateFlow<Address?> = _addressResult
+
+    private val _isActivate = MutableEventFlow<Boolean>()
+    val isActivate: EventFlow<Boolean> = _isActivate
+
+    override suspend fun handleEvent(event: Event) {
+        when (event) {
+            is AddressDetailEvent.SaveAddress -> {
+                saveAddress(event.address)
+            }
+
+            is AddressDetailEvent.CoordToAddress -> {
+                convertCoordToAddress(event.lat, event.lng)
+            }
+        }
+    }
+
+    private fun saveAddress(address: Address) {
+        DeliBuddyApplication.prefs.saveUserAddress(address)
+
+        // save address test
+        val test = DeliBuddyApplication.prefs.getCurrentUserAddress()
+        Timber.w("Save Success ${test.addressName}, lat: ${test.lat}, lon: ${test.lng}")
+    }
+
+    private fun convertCoordToAddress(lat: Double, lng: Double) {
+        job?.cancel()
+        job = viewModelScope.launch {
+            when (val result = coordToAddressUseCase(Pair<Double, Double>(lat, lng))) {
+                is NetworkResult.Success -> {
+                    Timber.w("success to convert")
+                    _addressResult.value = result.data
+                }
+
+                is NetworkResult.Error -> handleError(result) {
+
+                }
+            }
+        }
+    }
+
+    override suspend fun handleError(result: NetworkResult.Error, retryAction: RetryAction?) {
+        _isActivate.emit(false)
+        Timber.w("fail to convert")
+
+        when (result.errorType) {
+            is NetworkError.Unknown -> {
+                showToast("알 수 없는 에러가 발생했습니다.")
+            }
+            is NetworkError.Timeout -> {
+                showToast("타임 아웃 에러가 발생했습니다.")
+            }
+            is NetworkError.InternalServer -> {
+                showToast("내부 서버에서 오류가 발생했습니다.")
+            }
+            is NetworkError.BadRequest -> {
+                val code = (result.errorType as NetworkError.BadRequest).code
+                val msg = (result.errorType as NetworkError.BadRequest).message
+                showToast("에러 코드 ${code}, $msg")
+            }
+        }
+    }
+
+    override fun onCleared() {
+        job = null
```

Comment:
```suggestion
        job = null
        super.onCleared()
```
super.onCleared가 선언되어야 할거에요!
이부분은 이번에 베이스 수정때 처리 해 놨으니 추후에 수정해주시면 됩니다

Reply:
이걸 빼먹었네요! 다른 부분에서도 그렇게 한 것 같아서 같이 수정했습니당